gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1067,74 +1074,37 @@ def update_quality_check(self,
-
-        Parameters
-        ----------
-        original_dataset : xr.Dataset
-            dataset being appended to
-        append_times : list
-            list of times forming the time index of the append
-
-        Returns
-        -------
-        bool
-            whether the original and appending datasets form a contiguous time axis
-        """
-        # Check if first time in append times is one time step ahead of the last time in the original dataset
-        last_time_in_original = original_dataset[self.time_dim].values[-1]
-        if not self.is_time_contiguous(append_times[0], last_time_in_original):
-            return False
-        # Check if all times to be appended are contiguous with the prior time step in the append time range
-        return self.are_times_contiguous(append_times)
-
-    def is_time_contiguous(self, time: datetime.datetime, previous_time: datetime.datetime) -> bool:
-        """
-        Return false if a given time is out of order and/or does not follow the previous time, or falls outside of an acceptable range of timedeltas
-
-        Parameters
-        ----------
-        time
-            A datetime.datetime object representing the timestamp being checked
-        previous_time
-            A datetime.datetime object representing the prior timestamp
-
-        Returns
-        -------
-        bool
-            Returns False for any unacceptable timestamp, otherwise True
+    def are_times_in_expected_order(self, times: tuple[datetime.datetime], expected_delta: np.timedelta64) -> bool:
         """
-        if self.irregular_update_cadence():
-            if not self.irregular_update_cadence()[0] <= (time - previous_time) <= self.irregular_update_cadence()[1]:
-                self.warn(f"Time value {time} and previous time {previous_time} are not consecutive")
-                return False
-        elif time - previous_time != self.span_to_timedelta():
-            self.warn(f"Time value {time} and previous time {previous_time} are not consecutive")
-            return False
-        return True
-
-    def are_times_contiguous(self, times: tuple[datetime.datetime]) -> bool:
-        """
-        Convenience method to run `is_time_contiguous` in a loop over an ascending ordered list of input times,
-        since this is a regular pattern
+        Return false if a given iterable of times is out of order and/or does not follow the previous time,
+        or falls outside of an acceptable range of timedeltas
 
         Parameters
         ----------
         times
             A datetime.datetime object representing the timestamp being checked
+        expected_delta
+            Amount of time expected to be between each time
 
         Returns
         -------
         bool
-            Returns False for any unacceptable timestamp, otherwise True
+            Returns False for any unacceptable timestamp order, otherwise True
         """
+        # Warn if not using expected delta
```

Comment:
Related to the prev comment, I think splitting this into two functions would be better than printing a warning, maybe are_times_contiguous and are_times_in_cadence_order